ports/geckolib/glue.rs

```diff
@@ -355,10 +355,11 @@ pub extern "C" fn Servo_GetComputedValuesForPseudoElement(parent_style: *mut Ser
 pub extern "C" fn Servo_InheritComputedValues(parent_style: *mut ServoComputedValues)
      -> *mut ServoComputedValues {
     type Helpers = ArcHelpers<ServoComputedValues, ComputedValues>;
-    Helpers::with(parent_style, |parent| {
-        let style = ComputedValues::inherit_from(parent);
-        Helpers::from(style)
-    })
+    let style = match parent_style {
+        p if p.is_null() => Arc::new(ComputedValues::initial_values().clone()),
+        p => Helpers::with(p, |parent| { ComputedValues::inherit_from(parent) }),
```

Comment:
Why not:

```rust
let style = if parent_style.is_null() {
    Arc::new(...)
} else {
    Helpers::with(...)
};
```

I think that's a bit more straight-forward. Also, I'd personally remove the braces around the closure if it's single line, but it's fine if you don't want to.

Finally, repeating the question in the bugzilla bug... when does this happen? Is it expected or a workaround? A comment would be nice either here or in the caller.